designs/generic-git-provider-gitops.md

```diff
@@ -0,0 +1,205 @@
+gitproviderdesign.md
+
+# Generic Git Provider for EKS Anywhere GitOps Configuration
+## Introduction
+EKS Anywhere allows a user to employ a GitOps methodology to manage their clusters. This is achieved using Flux, an open and extensible set of continuous delivery solutions for Kubernetes.
+
+When a customer configures their cluster to use GitOps, they specify a Git provider to act as the source of truth for their cluster configuration. EKS Anywhere then bootstraps Flux onto the users cluster, and Flux is  configured to watch the user provided repository for changes to reconcile.
+
+At launch, EKS Anywhere officially supported cloud hosted GitHub as the only GitOps git hosting provider. This means that an EKS Anywhere user can configure their cluster to use Github as the source of truth for Flux. However, many customers use a variety of hosted and self-hosted Git providers, including but not limited to Github Enterprise, Gitlab, and Bitbucket. Rather than implement an EKS Anywhere git provider for each possible git hosting solution we want to enable customers across the spectrum of use-cases by offering them the ability to use a generic git provider as their source of truth for GitOps. 
+
+By 'generic git provider', we mean provider-agnostic basic authentication to a standard git repository via username/password and/or SSH.
```

Comment:
For GitHub we don't support username and password right? Is username/password something we want to address or should we just be okay with talking to repositories through only ssh.

Reply:
@vivek-koppuru from the discussion, are you going to check on the GitHub Enterprise use case and wether the PAT is being used (which would be used as a password in basic auth)?